rest-server/server/src/main/scala/com/actionml/authserver/services/CachedAuthorizationService.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright ActionML, LLC under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * ActionML licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.actionml.authserver.services
+
+import java.util.concurrent.TimeUnit
+
+import com.actionml.authserver.{AccessToken, ResourceId, RoleId}
+import com.typesafe.scalalogging.LazyLogging
+import org.ehcache.{CacheManager, ValueSupplier}
+import org.ehcache.config.builders.{CacheConfigurationBuilder, CacheManagerBuilder, ResourcePoolsBuilder}
+import org.ehcache.expiry.{Duration, Expirations, Expiry}
+import scaldi.{Injectable, Injector}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class CachedAuthorizationService(implicit inj: Injector) extends ClientAuthorizationService with LazyLogging with Injectable {
+  private implicit val _ = inject[ExecutionContext]
+  import CachedAuthorizationService.tokenCache
+
+  override def authorize(token: AccessToken, roleId: RoleId, resourceId: ResourceId): Future[Boolean] = {
+    val key = (token, roleId, resourceId)
+    if (tokenCache.containsKey(key)) {
+      Future.successful(tokenCache.get(key))
+    } else {
+      super.authorize(token, roleId, resourceId).map { result =>
+        tokenCache.put(key, result)
+        result
+      }
+    }
+  }
+}
+
+
+object CachedAuthorizationService {
+  val cacheTtl = Duration.of(30, TimeUnit.MINUTES)
```

Comment:
We should probably have all timeouts and limits in general defined in config with a method to override with harness-auth-env.sh or plain old env vars. This will clutter things but we can also choose to not add them to not put them in harness-auth-env.sh but they are there if we need them.

Reply:
I plan to deploy this Thursday so clean up anything you need to before then.